components/playlist/resources/reducers/playlist_reducer.ts

```diff
@@ -12,14 +12,30 @@ import * as PlaylistMojo from 'gen/brave/components/playlist/mojom/playlist.mojo
 const playlistReducer: Reducer<Playlist.State|undefined> =
     (state: Playlist.State|undefined, action) => {
       if (state === undefined) {
-        state = { lists: [] }
+        state = { lists: [], currentList: undefined }
       }
 
-      // TODO(sko) Handle each action properly.
       switch (action.type) {
         case types.PLAYLIST_LOADED:
-          state = { ...state, lists: [...action.payload as PlaylistMojo.Playlist[]] }
+          const playlists = action.payload
+          let currentList: PlaylistMojo.Playlist | undefined
+          if (state.currentList) {
+            // NOTE: win-x86 complains about this line.
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+            currentList = playlists.find((list: PlaylistMojo.Playlist) => list.id === state!.currentList!.id)
```

Comment:
Oh man, that's annoying :D I wonder if one of them is using a different version of Typescript?

Were you removing both `!`'s? I suspect the first one is triggering the `Assertion is unnecessary` (because state always exists) and the second one is necessary to not break mac (because currentList could be undefined?) 

Worst comes to worst, you could replace the `!` with `?`

Reply:
Yeah, both ! has same problem. Not sure what my ts version is. But CI pr builder has same problem so I guess it's not only my local problem. Like you said, `state` is assigned with new one so the first `!` isn't necessary 😹

> you could replace the ! with ?

This works for all platforms. I'll go with this and remove the line for suppressing linter :)